### PR TITLE
notes for 19858

### PR DESCRIPTION
**Pull request by amitiuttarwar**
for @jnewbery to review / edit / scrap / whatever

### CODE REVIEW

File: _posts/2020-11-25-#19858.md [hunk 1]
```diff
@@ -10,13 +10,47 @@ status: upcoming
 commit:
 ---
 
-_Notes and questions to follow soon!_
 
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
+- An [eclipse attack](https://bitcoinops.org/en/topics/eclipse-attacks/) is
+  when an adversary is able to isolate a victim from the bitcoin network by
```
Comment (jonatack) [inline _posts/2020-11-25-#19858.md:16]: s/bitcoin/Bitcoin/

can also link to https://bitcoincore.reviews/16702#eclipse-attack

File: _posts/2020-11-25-#19858.md [hunk 2]
```diff
@@ -10,13 +10,47 @@ status: upcoming
...
+  obfuscating network topology, they help mitigate eclipse attacks.
+
+- Block-relay-only connections were introduced in
+  [#15759](https://github.com/bitcoin/bitcoin/pull/15759). After these changes,
+  nodes by default open two outbound block-relay-only connections on startup.
+  We discussed this PR in a [previous review
+  club](https://bitcoincore.reviews/15759.html) if you are interested in
+  learning more.
+
+- This weeks PR, [#19858](https://github.com/bitcoin/bitcoin/pull/19858)
```
Comment (jonatack) [inline _posts/2020-11-25-#19858.md:30]: s/weeks/week's/ and add a comma after 19858 (or omit the first comma)

File: _posts/2020-11-25-#19858.md [hunk 3]
```diff
@@ -10,13 +10,47 @@ status: upcoming
...
-_Notes and questions to follow soon!_
 
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
+- An [eclipse attack](https://bitcoinops.org/en/topics/eclipse-attacks/) is
+  when an adversary is able to isolate a victim from the bitcoin network by
+  overtaking all of its connections.
+
+- Block-relay-only connections are a type of connection where two nodes do not
+  participate in transaction or address relay, and solely relay blocks. By
```
Comment (jonatack) [inline _posts/2020-11-25-#19858.md:20]: omit comma

File: _posts/2020-11-25-#19858.md [hunk 4]
```diff
@@ -10,13 +10,47 @@ status: upcoming
...
 
-_Notes and questions to follow soon!_
 
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
+- An [eclipse attack](https://bitcoinops.org/en/topics/eclipse-attacks/) is
+  when an adversary is able to isolate a victim from the bitcoin network by
+  overtaking all of its connections.
+
+- Block-relay-only connections are a type of connection where two nodes do not
```
Comment (jonatack) [inline _posts/2020-11-25-#19858.md:19]: maybe omit "two" unless additional info is added, as it's not clear from the preceding text where the number comes from and this sententce describes what block-relay-only connections are and not the implementation

File: _posts/2020-11-25-#19858.md [hunk 5]
```diff
@@ -10,13 +10,47 @@ status: upcoming
...
+- After the node opens a block-relay-only connection to a peer, it begins to
+  sync headers. If this reveals new blocks, the eviction logic will rotate out
+  an existing block-relay-only connection. If no new blocks are discovered, the
+  connection is closed. It's important for this eviction logic to be
+  carefully reviewed.
+
 
 ## Questions
--->
+1. What is a scenario under which periodic block-relay-only connections could
```
Comment (jonatack) [inline _posts/2020-11-25-#19858.md:43]: maybe s/under/in/

**Review by jonatack — COMMENTED** [review]
Great choice of PR!

**Comment by jonatack** [comment]
Excellent questions.

**Comment by jonatack** [comment]
Excellent questions. Thanks for writing this, @amitiuttarwar.

File: _posts/2020-11-25-#19858.md (hunk 1 above)
Comment (jnewbery) [reply _posts/2020-11-25-#19858.md:16]: Done!

File: _posts/2020-11-25-#19858.md (hunk 3 above)
Comment (jnewbery) [reply _posts/2020-11-25-#19858.md:20]: done

File: _posts/2020-11-25-#19858.md (hunk 2 above)
Comment (jnewbery) [reply _posts/2020-11-25-#19858.md:30]: done

File: _posts/2020-11-25-#19858.md (hunk 4 above)
Comment (jnewbery) [reply _posts/2020-11-25-#19858.md:19]: done

File: _posts/2020-11-25-#19858.md (hunk 5 above)
Comment (jnewbery) [reply _posts/2020-11-25-#19858.md]: done

**Comment by jnewbery** [comment]
Thanks for the excellent notes & questions @amitiuttarwar and the review @jonatack.

**Pull request merged**